From: jbeulich@suse.com
Subject: xenbus: correctly signal errors from xenstored_local_init()
References: luckily none so far
Patch-mainline: Never, SUSE-Xen specific

The factoring out of that code had lost the -ENOMEM error indicator.
Dropy a nearby pointless initializer at once.

Reported-by: Pan Bian <bianpan2016@163.com>

--- a/drivers/xen/xenbus/xenbus_probe.c
+++ b/drivers/xen/xenbus/xenbus_probe.c
@@ -1221,8 +1221,8 @@ fail0:
  */
 static int __init xenstored_local_init(void)
 {
-	int err = 0;
-	unsigned long page = 0;
+	int err = -ENOMEM;
+	unsigned long page;
 	struct evtchn_alloc_unbound alloc_unbound;
 
 	/* Allocate Xenstore page */
